pkg/collector/stats/node_stats.go

```diff
@@ -19,18 +19,175 @@ package stats
 import (
 	"fmt"
 
+	"math"
+
 	"github.com/sustainable-computing-io/kepler/pkg/config"
 	"github.com/sustainable-computing-io/kepler/pkg/node"
 	acc "github.com/sustainable-computing-io/kepler/pkg/sensors/accelerator"
 	"github.com/sustainable-computing-io/kepler/pkg/utils"
+	"k8s.io/klog/v2"
 )
 
+var (
+	// Modify Manually
+	spreadDiff           = 0.3
+	historyLength        = 10
+	energyTypeToMinSlope = map[string]float64{
+		config.IdleEnergyInCore:   0.01,
+		config.IdleEnergyInPkg:    0.3,
+		config.IdleEnergyInUnCore: 0.3,
+		config.IdleEnergyInDRAM:   0.3,
+	}
+	energyTypeToMinIntercept = map[string]float64{
+		config.IdleEnergyInCore:   0,
+		config.IdleEnergyInPkg:    0,
+		config.IdleEnergyInUnCore: 0,
+		config.IdleEnergyInDRAM:   0,
+	}
+)
+
+type IdleEnergyResult struct {
+	calculatedIdleEnergy float64
+	diff                 float64
+	history              []float64
+}
+
+type IdleEnergyCalculator struct {
```

Comment:
@KaiyiLiu1234 , could you please add a unit test for this? It will help us understand the behaviour better.

Reply:
Understood. I can add this unit test in node_stats_test.go